### PR TITLE
BCMOHAD 24464: Status Tracker ETL Updates

### PR DESCRIPTION
**Pull request by npham49**
- Updates to the Status Tracker ETL to follow the new Data Dictionary.
- Added helper functions for Notes and Date fields to the Root model.
- Re-format PCDConstants.

### CODE REVIEW

**Review by weskubo-cgi — CHANGES_REQUESTED** [review]

File: src/main/java/ca/bc/gov/chefs/etl/forms/pcd/statusTracker/processor/StatusTrackerFormApiResponseProcessor.java [hunk 1]
```diff
@@ -19,20 +20,29 @@
...
+
 	private static final String DEFAULT_EMAIL = "pcdbi-status-tracker-data-loader@gov.bc.ca";
-	
-    private static final String INITIATIVE_TYPE_PCN = "PCN";
+
+	private static final String INITIATIVE_TYPE_PCN = "PCN";
+	private static final String INITIATIVE_TYPE_CHC = "CHC";
+	private static final String INITIATIVE_TYPE_FNPCC = "FNPCC";
+	private static final String INITIATIVE_TYPE_NPPCC = "NPPCC";
+	private static final String INITIATIVE_TYPE_UPCC = "UPCC";
```
[the remark this answers is not in the document]
Comment (npham49) [reply src/main/java/ca/bc/gov/chefs/etl/forms/pcd/statusTracker/processor/StatusTrackerFormApiResponseProcessor.java]: I've moved these into the main PCDConstants in the latest commit.

File: src/main/java/ca/bc/gov/chefs/etl/forms/pcd/statusTracker/processor/StatusTrackerFormApiResponseProcessor.java [hunk 2]
```diff
@@ -44,113 +54,362 @@ public void process(Exchange exchange) throws Exception {
...
+					StatusTrackerSubmission::setSubmissionVersion);
+			mapper.map(src -> src.getForm().getFormName(),
+					StatusTrackerSubmission::setSubmissionFormName);
+			mapper.map(src -> src.getTargetLaunchOpenDate(),
+					StatusTrackerSubmission::setTargetOpeningDate);
+			mapper.map(src -> src.getActualLaunchOpenDate(),
+					StatusTrackerSubmission::setActualOpeningDate);
+		});
+
+		ModelMapper InitiativeMapper = new ModelMapper();
```
[the remark this answers is not in the document]
Comment (npham49) [reply src/main/java/ca/bc/gov/chefs/etl/forms/pcd/statusTracker/processor/StatusTrackerFormApiResponseProcessor.java]: This naming is updated in the latest commit to be lowercase.

File: src/main/java/ca/bc/gov/chefs/etl/forms/pcd/statusTracker/processor/StatusTrackerFormApiResponseProcessor.java [hunk 3]
```diff
@@ -44,113 +54,362 @@ public void process(Exchange exchange) throws Exception {
...
+								mapper.map(src -> src.getPcnPlSerPlnOprBudEdrNotes(),
+										PCNStatusTrackerItem::setPcnPlSerPlnEdrNotes);
+								mapper.map(src -> src.getPcnPlSerPlnOprBudSubUndRevDate(),
+										PCNStatusTrackerItem::setPcnPlSerPlnRevDate);
+								mapper.map(src -> src.getPcnPlSerPlnOprBudSubUndRevNotes(),
+										PCNStatusTrackerItem::setPcnPlSerPlnRevNotes);
+							});
+					PCNStatusTrackerItem pcnStatusTrackerItem =
+							InitiativeMapper.map(root, PCNStatusTrackerItem.class);
+					pcnStatusTrackerItem.setPcnId(java.util.UUID.randomUUID().toString());
```
[the remark this answers is not in the document]
Comment (npham49) [reply src/main/java/ca/bc/gov/chefs/etl/forms/pcd/statusTracker/processor/StatusTrackerFormApiResponseProcessor.java]: This is now fixed to UUID.randomUUID().toString().

File: src/main/java/ca/bc/gov/chefs/etl/forms/pcd/statusTracker/processor/StatusTrackerFormApiResponseProcessor.java [hunk 4]
```diff
@@ -44,113 +54,362 @@ public void process(Exchange exchange) throws Exception {
...
+								mapper.map(src -> src.getChcPlFunPkgAppIssNotes(),
+										CHCStatusTrackerItem::setChcPlFunPkgAppNotes);
+								mapper.map(src -> src.getChcPlFunPkgAppPenDate(),
+										CHCStatusTrackerItem::setChcPlFunPkgPenDate);
+								mapper.map(src -> src.getChcPlFunPkgAppPenNotes(),
+										CHCStatusTrackerItem::setChcPlFunPkgPenNotes);
+							});
+					CHCStatusTrackerItem chcStatusTrackerItem =
+							InitiativeMapper.map(root, CHCStatusTrackerItem.class);
+					chcStatusTrackerItem.setChcId(java.util.UUID.randomUUID().toString());
```
[the remark this answers is not in the document]
Comment (npham49) [reply src/main/java/ca/bc/gov/chefs/etl/forms/pcd/statusTracker/processor/StatusTrackerFormApiResponseProcessor.java]: This is now fixed to UUID.randomUUID().toString().

File: src/main/java/ca/bc/gov/chefs/etl/forms/pcd/statusTracker/processor/StatusTrackerFormApiResponseProcessor.java [hunk 5]
```diff
@@ -44,113 +54,362 @@ public void process(Exchange exchange) throws Exception {
...
+										UPCCStatusTrackerItem::setUpccPlSerPlnEdrNotes);
+								mapper.map(src -> src.getUpccPlSerPlnOprBudSubUndRevDate(),
+										UPCCStatusTrackerItem::setUpccPlSerPlnRevDate);
+								mapper.map(src -> src.getUpccPlSerPlnOprBudSubUndRevNotes(),
+										UPCCStatusTrackerItem::setUpccPlSerPlnRevNotes);
+							});
+					UPCCStatusTrackerItem upccStatusTrackerItem =
+							InitiativeMapper.map(root, UPCCStatusTrackerItem.class);
+
+					upccStatusTrackerItem.setUpccId(java.util.UUID.randomUUID().toString());
```
[the remark this answers is not in the document]
Comment (npham49) [reply src/main/java/ca/bc/gov/chefs/etl/forms/pcd/statusTracker/processor/StatusTrackerFormApiResponseProcessor.java]: This is now fixed to UUID.randomUUID().toString().

File: src/main/java/ca/bc/gov/chefs/etl/forms/pcd/statusTracker/processor/StatusTrackerFormApiResponseProcessor.java [hunk 6]
```diff
@@ -44,113 +54,362 @@ public void process(Exchange exchange) throws Exception {
...
+								mapper.map(src -> src.getFnpccPlSerPlnOprBudSubUndRevNotes(),
+										FNPCCStatusTrackerItem::setFnpccPlSerPlnRevNotes);
+								mapper.map(src -> src.getFnpccPlTrgSerPlnRevDate(),
+										FNPCCStatusTrackerItem::setFnpccPlTrgPlnRevDate);
+								mapper.map(src -> src.getFnpccPlTrgSerPlnRevNotes(),
+										FNPCCStatusTrackerItem::setFnpccPlTrgPlnRevNotes);
+							});
+					FNPCCStatusTrackerItem fnpccStatusTrackerItem =
+							InitiativeMapper.map(root, FNPCCStatusTrackerItem.class);
+					fnpccStatusTrackerItem.setFnpccId(java.util.UUID.randomUUID().toString());
```
[the remark this answers is not in the document]
Comment (npham49) [reply src/main/java/ca/bc/gov/chefs/etl/forms/pcd/statusTracker/processor/StatusTrackerFormApiResponseProcessor.java]: This is now fixed to UUID.randomUUID().toString().

File: src/main/java/ca/bc/gov/chefs/etl/forms/pcd/statusTracker/processor/StatusTrackerFormApiResponseProcessor.java [hunk 7]
```diff
@@ -44,113 +54,362 @@ public void process(Exchange exchange) throws Exception {
...
+					modelMapper.map(root, StatusTrackerSubmission.class);
+
 			// Default the email for bulk uploaded records
 			if (StringUtils.isBlank(hiStatusTracker.getSubmitterEmail())) {
 				hiStatusTracker.setSubmitterEmail(DEFAULT_EMAIL);
 			}
-			
+
+			// Set the values not parsed by ModelMapper
+			hiStatusTracker.setLateEntry(root.getLateEntry());
```
[the remark this answers is not in the document]
Comment (npham49) [reply src/main/java/ca/bc/gov/chefs/etl/forms/pcd/statusTracker/processor/StatusTrackerFormApiResponseProcessor.java]: I had some issues with Boolean values in ModelMapper, but this seems to be fixed now, and the output is the same with or without this assignment. Hence, I've removed it in the latest commit.

File: src/main/java/ca/bc/gov/chefs/etl/forms/pcd/statusTracker/processor/StatusTrackerFormApiResponseProcessor.java [hunk 8]
```diff
@@ -44,113 +54,362 @@ public void process(Exchange exchange) throws Exception {
...
+										PCNStatusTrackerItem::setPcnPlSerPlnEdrNotes);
+								mapper.map(src -> src.getPcnPlSerPlnOprBudSubUndRevDate(),
+										PCNStatusTrackerItem::setPcnPlSerPlnRevDate);
+								mapper.map(src -> src.getPcnPlSerPlnOprBudSubUndRevNotes(),
+										PCNStatusTrackerItem::setPcnPlSerPlnRevNotes);
+							});
+					PCNStatusTrackerItem pcnStatusTrackerItem =
+							InitiativeMapper.map(root, PCNStatusTrackerItem.class);
+					pcnStatusTrackerItem.setPcnId(java.util.UUID.randomUUID().toString());
+					pcnStatusTrackerItem.setAllClinicsImpacted(root.getAllClinicsImpacted());
```
[the remark this answers is not in the document]
Comment (npham49) [reply src/main/java/ca/bc/gov/chefs/etl/forms/pcd/statusTracker/processor/StatusTrackerFormApiResponseProcessor.java]: Same has the lateEntry above, but this is now working properly so I'll remove this line in the latest commit.

**Review by weskubo-cgi — APPROVED** [review]

**Pull request merged**